docs/troubleshooting.md

```diff
@@ -18,17 +18,118 @@ might overload systems if left on indefinitely.
 Spans might be dropped, or lost, due to several reasons. Follow these steps to ensure
 spans are not being dropped by the `SDK`.
 
-1. [Enable debug logging](#enable-debug-logging). This logs the `SDK`
-   configuration and all spans sent to registered `SpanProcessor`s. Verify the
-   that spans are exported in the log messages. Check also the`total_dropped` count.
-2. If an incorrect number of spans are being reported as exported, verify you
-   are ending the spans you expect to see.
-3. If spans are reported as dropped, the `BatchSpanProcessor` is dropping
-   spans. Increase the capacity of the `BatchSpanProcessor`'s queue or the
-   frequency it exports via [environment variables].
-
-Following the previous steps might help understand if spans are being dropped in the
-`SDK` or downstream.
+### Missing all span from a service
+
+If you see no spans in the Splunk Observability Platform for your service
+ensure the following first.
+
+1. Make sure to wait a few minutes. There may be delays in the telemetry
+   pipeline. Wait a few minutes and double check.
+2. Ensure your service is named properly. Check that your service names are not
+   showing up in the Splunk Observability Cloud interface with a prefix of
+   `unknown_service:` (i.e. `unknown_service:go` or
+   `unknown_service:<your-programs-binary-name>`). If so, set the
+   `OTEL_SERVICE_NAME` environment variable to your service's name and restart
+   your application.
+
+If you have taken these steps and still are not able to see spans, [enable
+debug logging](#enable-debug-logging). This should produce a log message like
+the following:
+
+```text
+debug	global/internal_logging.go:62	exporting spans	{"count": 154, "total_dropped": 0}
+```
+
+The `count` in the log message is the number of spans that were exported for a
+batch by the `SDK`.
+
+If `count` is non-zero that means the `SDK` is exporting span. If this is the
+case, spans are being dropped downstream. Refer to [the collector
+troubleshooting documentation].
+
+If `count` is zero, that means the `SDK` is not exporting any spans. If this is
+the case, ensure that all spans your application is creating are ended (i.e.
+`span.End()`).
+
+The `total_dropped` value of the log message is the cumulative number of spans
+the `SDK` has dropped. If this value is non-zero see the [missing some spans
+from a service](#missing-some-spans-from-a-service) section for more
+information on how to resolve this.
+
+### Missing some spans from a service
+
+If you see traces from your service in the Splunk Observability Platform
+missing spans you likely need to configure the `BatchSpanProcessor` batching
+configuration. First verify spans are being dropped by [enabling debug
+logging](#enable-debug-logging). This should produce a log message like the
+following:
```

Comment:
```suggestion
configuration. Verify whether spans are being dropped by [enabling debug
logging](#enable-debug-logging). This might produce a log message like the
following:
```